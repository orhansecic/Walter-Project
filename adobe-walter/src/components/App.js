import React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {connect} from 'react-redux';
import {BrowserRouter, Route} from 'react-router-dom';

//Components
import Header from './Header.js';
import Invoices from './All.js';
import Pending from './Pending.js';
import HeaderMain from './main/HeaderMain.js';
import Approved from './Approved.js';
import Popup from './Popup.js';

import {invoicesActionCreator, approvedActionCreator, pendingActionCreator} from '../actions/index.js';

import '../styles/style.css';

class App extends React.Component{
    createMockInstance(){
        const mock = new MockAdapter(axios, {delayResponse: 4000});
        mock.onGet("/invoices").reply(200, {
            invoices: [
                {id: 1, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Approved"},
                {id: 2, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Jane Doe", cost: "120 KM", status: "Approved"},
                {id: 3, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Sugar Ray", cost: "250 KM", status: "Approved"},
                {id: 4, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Joe Rogan", cost: "20 KM", status: "Approved"},
                {id: 5, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "G.I. Joe", cost: "185 KM", status: "Approved"},
                {id: 6, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 7, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 8, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 9, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 10, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 11, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Declined"},
                {id: 12, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Declined"},
                {id: 13, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Declined"},
                {id: 14, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Declined"},
                {id: 15, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Declined"},
            ]
        })

        mock.onGet("/approved").reply(200,{
            invoices:[
                {id: 1, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Approved"},
                {id: 2, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Jane Doe", cost: "120 KM", status: "Approved"},
                {id: 3, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Sugar Ray", cost: "250 KM", status: "Approved"},
                {id: 4, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "Joe Rogan", cost: "20 KM", status: "Approved"},
                {id: 5, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "G.I. Joe", cost: "185 KM", status: "Approved"},
            ]
        })

        mock.onGet("/pending").reply(200,{
            invoices:[
                {id: 6, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 7, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 8, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 9, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
                {id: 10, invoiceName: "Invoice name.pdf", date: "04/06/2020", issuedBy: "John Doe", cost: "200 KM", status: "Pending"},
            ]
        })
    }
    componentDidMount(){
        this.createMockInstance();
    }
    
    render(){
        return(
                <BrowserRouter>
                    <Route path="/" component={Header}></Route>
                    <Route path="/" component={HeaderMain}></Route>
                    <Route path="/" exact component={Invoices}></Route>
                    <Route path="/pending" component={Pending}></Route>
                    <Route path="/approved" component={Approved}></Route>
                    <Route path="/example" component={Popup}></Route>
                </BrowserRouter>
        );
    }
}

const mapStateToProps = (state) =>{
    return{
        invoices: state.invoices,
    }
}

export default connect (mapStateToProps,
    {
        invoicesActionCreator: invoicesActionCreator,
        approvedActionCreator: approvedActionCreator,
        pendingActionCreator: pendingActionCreator,
    }) (App);