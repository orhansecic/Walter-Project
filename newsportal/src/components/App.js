import React from 'react';
import {BrowserRouter, Route} from 'react-router-dom';

//Components
import LandingPage from './LandingPage.js';
import Article from './Article.js';

const App = () =>{
    return(
        <div className="ui container">
            <BrowserRouter>
                <Route path="/" exact component={LandingPage}></Route>
                <Route path="/article" component={Article}></Route>
            </BrowserRouter>
        </div>
        
    );
};

export default App;