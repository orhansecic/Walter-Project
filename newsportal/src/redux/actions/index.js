// Action Creators and Actions with Thunk

//Axios
import newsAPI from '../../api/newsAPI.js';

const topHeadlinesAction = () =>{
    return  function (dispatch, getState){
        newsAPI.get("/top-headlines",{
            params:{
                country: "us",
            }
        })
        .then((value) => {
            dispatch({
                type: "TOP_HEADLINES",
                payload: value.data.articles,
            })
        })
        .catch((err) => {
            console.error(err);
        })
    }
};

export {topHeadlinesAction};

const selectedArticleAction = (article) =>{
    return {
        type: "SELECTED_ARTICLE",
        payload: article,
    }
};

export {selectedArticleAction};

const searchedArticlesAction = (term, orderValue="publishedAt") =>{
    return function (dispatch, getState){
        return newsAPI("/everything",{
            params:{
                q: term,
                sortBy: orderValue,
            },
        })
        .then((value) => {
            dispatch({
                type: "SEARCHED_ARTICLES",
                payload: value.data.articles,
            })
        })
        .catch((err) => {
            console.error(err);
        })
    } 
    
}

export {searchedArticlesAction};
