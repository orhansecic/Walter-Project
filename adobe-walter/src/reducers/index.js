import produce from 'immer';
// Reducers
import { combineReducers } from 'redux';

const invoiceReducer = (invoices = [], action) =>{
    switch(action.type){
        case "GET_INVOICES":
            return action.payload;

        case "APPROVED_INVOICES":
            return action.payload;

        case "PENDING_INVOICES":
            return action.payload;
        
         case "NEW_INVOICE":
            return produce(invoices, (draftState) => {
                draftState.push(action.payload);
            })
            //return [...invoices, action.payload];
        case "SORT_INVOICES":
            return action.payload;

        default: 
            return invoices;
    }
};

const isOpenReducer = (isOpen = false, action) =>{
    switch(action.type){
        case "ISOPEN_TRUE":
            return action.payload;
        case "ISOPEN_FALSE":
            return action.payload;
        default:
            return isOpen;
    }
}

const isPending = (state = false, action) =>{
    switch(action.type){
        case "SET_PENDING":
            return action.payload;
        default:
            return state;
    }
}

const reducers = combineReducers({
    invoices: invoiceReducer,
    isOpen: isOpenReducer,
    pending: isPending,
});

export default reducers;