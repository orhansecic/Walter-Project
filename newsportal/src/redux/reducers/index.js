// Reducers
import {combineReducers} from 'redux';

const topHeadlinesReducer = (topHeadlines = [], action) =>{
    switch(action.type){
        case "TOP_HEADLINES":
            return action.payload;
        default:
            return topHeadlines;
    }
};

const selectedArticleReducer = (article = null, action) =>{
    switch(action.type){
        case "SELECTED_ARTICLE":
            return action.payload;
        default: 
            return article;
    }
}

const reducers = combineReducers({
    topHeadlines : topHeadlinesReducer,
    article: selectedArticleReducer,
})

export default reducers;