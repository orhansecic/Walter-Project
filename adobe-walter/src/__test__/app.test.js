import React from 'react';
import {Provider} from 'react-redux';
import {applyMiddleware, createStore} from 'redux';
import reducers from '../reducers';
import thunk from 'redux-thunk';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';

import App from '../components/App';

const renderWithRedux = (
    component,
    { initialState, store = createStore(reducers, initialState) } = {}
  ) => {
    return {
      ...render(<Provider store={store}>{component}</Provider>),
      store,
    }
  }

afterEach(cleanup);

describe("Snapshot Testing", () => {
    test("App Snapshot", () => {
        const{asFragment} = renderWithRedux(<App/>)

        expect(asFragment(<App/>)).toMatchSnapshot();
    })
});

describe("Query Testing", () => {
    test("Query Test 1", () =>{
        const {getByTestId} = render(<Provider store={createStore(reducers, applyMiddleware(thunk))}>
            <App/>
        </Provider>);

    expect(getByTestId("Example")).toHaveTextContent("Requests");
    })
})

describe("React Redux Test", () => {
    test("React Redux", () => {
        const {getByTestId} = render(<Provider store={createStore(reducers, applyMiddleware(thunk))}>
            <App/>
        </Provider>);

    })
})