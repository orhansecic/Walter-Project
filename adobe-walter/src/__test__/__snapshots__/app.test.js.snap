// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing App Snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="nav"
    >
      <div
        class="item item--left"
      >
        <a
          class="item__element element--logo"
          href="/"
        >
          <img
            alt="walter-logo"
            src="logo.png"
          />
        </a>
        <a
          class="item__element item__element--active element--requests"
          data-testid="Example"
          href="/"
        >
          Requests
        </a>
        <a
          class="item__element element--collaborators"
          href="/"
        >
          Collaborators
        </a>
      </div>
      <div
        class="item item--right"
      >
        <span
          class="item__element item__element--bell"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z"
            />
          </svg>
        </span>
        <svg
          class="item__element--icon"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
          />
        </svg>
        <span
          class="item__element"
        >
          Admin Name
        </span>
        <svg
          class="item__element"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 15l-4.243-4.243 1.415-1.414L12 12.172l2.828-2.829 1.415 1.414z"
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      class="nav-mobile"
    >
      <a
        class="item__element element--logo"
        href="/"
      >
        <img
          alt="walter-logo"
          src="logo.png"
        />
      </a>
      <a
        class="item__element item__element--active element--requests"
        href="/"
      >
        Requests
      </a>
      <a
        class="item__element element--collaborators"
        href="/"
      >
        Collaborators
      </a>
      <span
        class="item__element item__element--bell element--bell"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z"
          />
        </svg>
      </span>
      <span
        class="item__element element--admin"
      >
        Admin Name
      </span>
    </div>
  </div>
  <div
    class="header-main"
  >
    <div
      class="header-main__item"
    >
      <h3>
        Invoices
      </h3>
    </div>
    <div
      class="header-main__item"
    >
      <button
        class="item__primButton"
      >
        New Invoice
      </button>
      <input
        class="item__input"
        placeholder="Search Invoices"
      />
    </div>
  </div>
  <div>
    <div
      class="content"
    >
      <div
        class="sort"
      >
        <div
          class="sort__div1"
        >
          <a
            class="sort__item sort__item content--underline content--active"
            href="/"
          >
            All
          </a>
          <a
            class=" sort__item"
            href="/pending"
          >
            Pending
          </a>
          <a
            class="sort__item"
            href="/approved"
          >
            Approved
          </a>
        </div>
        <div
          class="sort__div2"
        >
          <span
            class="sort__item--right"
          >
            <span
              class="sort__item--faded"
            >
              Sort By: 
            </span>
            Date
          </span>
          <span
            class="sort__item--right"
          >
            <span
              class="sort__item--faded"
            >
              Sort By: 
            </span>
            Cost
          </span>
          <span
            class="sort__item--right"
          >
            <span
              class="sort__item--faded"
            >
              Showing 
            </span>
             0
          </span>
        </div>
      </div>
      <div
        style="text-align: center;"
      >
        <h1>
          The Content is Loading...
        </h1>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
